app/views/core/SubscribeModal.coffee

```diff
@@ -17,34 +17,47 @@ module.exports = class SubscribeModal extends ModalView
 
   events:
     'click #close-modal': 'hide'
-    'click .popover-content .parent-send': 'onClickParentSendButton'
-    'click .email-parent-complete button': 'onClickParentEmailCompleteButton'
     'click .purchase-button': 'onClickPurchaseButton'
     'click .stripe-lifetime-button': 'onClickStripeLifetimeButton'
     'click .back-to-products': 'onClickBackToProducts'
 
   constructor: (options={}) ->
     if document.location.host is 'br.codecombat.com'
       document.location.href = 'http://codecombat.net.br/'
-    
+
     super(options)
     @state = 'standby'
+    @couponID = utils.getQueryVariable('coupon')
+    @subType = utils.getQueryVariable('subtype', 'both-subs')
     if options.products
       # this is just to get the test demo to work
       @products = options.products
       @onLoaded()
     else
       @products = new Products()
       data = {}
-      if utils.getQueryVariable('coupon')?
-        data.coupon = utils.getQueryVariable('coupon')
+      if @couponID
+        data.coupon = @couponID
       @supermodel.trackRequest @products.fetch {data}
     @trackTimeVisible({ trackViewLifecycle: true })
     payPal.loadPayPal().then => @render()
-    @subType = utils.getQueryVariable('subtype', me.getSubModalGroup())
 
   onLoaded: ->
     @basicProduct = @products.getBasicSubscriptionForUser(me)
+    # Process basic product coupons unless custom region pricing
+    if @couponID and @basicProduct.get('coupons')? and @basicProduct?.get('name') is 'basic_subscription'
+      @basicCoupon = _.find(@basicProduct.get('coupons'), {code: @couponID})
+      # Always use both-subs UX test group when basic product coupon, and delay identify until we can decide
+      @subType = if utils.getQueryVariable('subtype')?
+        me.getSubModalGroup(utils.getQueryVariable('subtype'))
```

Comment:
This seems like this and the other calls in this function should be 'setSubModalGroup`, as its purpose is to set `User.subModalGroup`? I couldn't find any instances of it being used anyway, do we even need that prop?

Reply:
We have to force a specific test group in this scenario, and I want the analytics to be correct once we know which group we'll need.